examples/keras_imagenet_resnet50.py

```diff
@@ -31,6 +31,8 @@
                     help='tensorboard log directory')
 parser.add_argument('--checkpoint-format', default='./checkpoint-{epoch}.h5',
                     help='checkpoint file format')
+parser.add_argument('--fp16', action='store_true', default=False,
```

Comment:
Users will likely expect `--fp16` to mean FP16 computation.  Let's rename to `--fp16_allreduce`?